### ISSUE TITLE
backupccl: node restart during backup causes backup to fail

### ISSUE THREAD

**Issue by pbardea** [cockroachdb/cockroach#57635]
Since backup now distributes work to all nodes in a cluster up front and relies on the execution of the distsql flow, if a node were to become unavailable during a backup the flow will return a connection error and fail the backup.

Backup should be able to detect these kinds of errors, and return a JobsRetry error that will inform the jobs registry to not mark the job as failed, but retry adopting the job. With this scheme, if a node failure were to occur, the backup job would be retried and a new flow would be planned on all available nodes.

**Comment by pbardea** [comment]
We should also add version-upgrade roachtests that perform a rolling upgrade of the nodes. An interesting test case is upgrading/restarting the nodes which are not coordinating the backup.

**Comment by pbardea** [comment]
We improved this situation in https://github.com/cockroachdb/cockroach/pull/60947 where backup, restore and import jobs that have a flow will replan their flow on a set of "communication" errors. The retry policy only retries a fixed amount of times since we don't have full confidence in the error detection that the error is indeed transient. If we retry more than the specified amount of times we fail the job to surface the issue to the operator. This seems like a reasonable trade-off for now. Closing this issue until there's a more actionable goal to tackle here.
